packages/wallets/src/evm/wallets/abstract.ts

```diff
@@ -46,11 +47,29 @@ export async function checkContractWalletSignature(
   }
 }
 
+type AbstractWalletParams = {
+  apiKey?: string;
+};
+
 export abstract class AbstractWallet
   extends EventEmitter<WalletEvents>
   implements GenericAuthWallet, EVMWallet
 {
   public type: Ecosystem = "evm";
+  private params: AbstractWalletParams;
+
+  constructor(params: AbstractWalletParams) {
+    super();
+
+    if (!params.apiKey) {
+      console.warn(
+        "No API key provided. You will have limited access to thirdweb's services for storage, RPC, and account abstraction. You can get an API key from https://thirdweb.com/dashboard/",
```

Comment:
not going to warn here since it's not clear why you need it in wallets (yet)

removing this warn